src/features/renameVariableParts.ts

```diff
@@ -167,47 +178,53 @@ export const registerRenameVariableParts = () => {
                 moveVariableParts('down')
             }),
             registerCommand('renameVariablePartsAccept', async () => {
-                quickPick.hide()
                 const edit: vscode.WorkspaceEdit = await vscode.commands.executeCommand(
                     'vscode.executeDocumentRenameProvider',
                     document.uri,
                     selection.end,
                     getResultingName(),
                 )
                 await vscode.workspace.applyEdit(edit)
+                mainDisposable.dispose()
             }),
             registerCommand('renameVariablePartsAcceptReplace', () => {
                 void activeEditor.edit(builder => {
                     builder.replace(editRange, getResultingName())
                 })
-                quickPick.hide()
+                mainDisposable.dispose()
             }),
             {
                 dispose() {
                     void vscode.commands.executeCommand('setContext', 'zardoyExperiments.renameVariablePartsOpened', false)
                 },
             },
         )
-        quickPick.onDidAccept(() => {
+        quickPick.onDidAccept(async () => {
             const activeItem = quickPick.activeItems[0]!
-            if (editingIndex === undefined) {
-                if (!activeItem) return
-                editingIndex = quickPick.items.indexOf(activeItem)
-                const { label } = activeItem
-                quickPick.items = []
-                quickPick.title = `Editing part: ${getQuickPickTitle()}`
-                quickPick.value = label
-                return
+            if (!activeItem) return
+            const updatingPartIndex = quickPick.items.indexOf(activeItem)
+            const { label } = activeItem
+            isDisposeEnabled = false
+            const renamedPart = await vscode.window.showInputBox({ value: label, title: updateMainTitle('input') })
+
+            if (renamedPart) {
+                parts.splice(updatingPartIndex, 1, renamedPart)
+                updateQuickPick()
+                setActiveItem(updatingPartIndex)
```

Comment:
Right now if the user presses `esc` selected part index is dropped. Not sure if we should preserve it, but, anyway, implementation is pretty simple if needed